k0789885/MyGame.py

```diff
@@ -1,239 +1,202 @@
-    BarCh = ""
-    while BarCh != 1 and BarCh != 2 and BarCh != 3:
-        print "What do you do?"
-        print "1. Bartender"
-        print "2. Maidens"
-        print "3. Cloaked Man"
-        print "4. Leave"
-        BarCh = raw_input("> ")
-        if BarCh == "1":
-            bartender()
-        elif BarCh == "2":
-            maidens()
-        elif BarCh == "3":
-            print "(The air around him is grim and fills you with doubt.)"
-            cloaked_man()
-        elif BarCh == "4":
-            arrivetown()
-        else:
-            notvalid()
-            
-def arrivetown():
-    print "There is a bar, church, and a few homes. What would you like to go?"
-    print "1. Bar"
-    print "2. Church"
-    print "3. Homes"
-    print "4. Leave"
-    choice = ""
-    while choice != 1 and choice != 2 and choice != 3:
-        choice = raw_input("> ")
-        if choice == "1":
-            print "You push the doors open and are greeted to a room of drunkards and maidens."
-            bar()
-        elif choice == "2":
-            print "You walk into the church. The clergy seems unrested and the Father of the church is speaking to nuns."
-            print "What do you do?"
-            gochurch()
-        elif choice == "3":
-            print "(The houses consist of one large home and three small homes)"
-            homes()
-        elif choice == "4":
-            print "You left the town to it's own fate, no need to get involved."
-            print ""
-            gameover()
-        else:
-            notvalid()
+class Menu(object):
+
+   def __init__(self):
+       self.choices = []
+       self.texts = []
+       self.selected_value = None
+       self.invalid_text = "That's not one of the choices, please use one of the numbers displayed."
+
+   def add_choice(self, display_text, value):
+       self.choices.append(value)
+       self.texts.append(display_text)
+   
+   def current_weapon(self, weapon):
+       self.current_weapon = weapon.selected_value
+
+   def prompt(self, text):
+
+       while self.selected_value is None:
+           print text
+           num = 1
+           for display_text in self.texts:
+               print "%d. %s" % (num, display_text)
+               num = num + 1
+
+           user_choice = int(raw_input("> "))
+
+           if user_choice < 1 or user_choice > len(self.texts):
+               print self.invalid_text
+           else:
+               self.selected_value = self.choices[user_choice - 1]
             
-#((((Start of Game))))
+               
+#((((Sparing Pit 1 / Weapon Select))))
 
-while Class != "Warrior" and Class != "Archer" and Class != "Mage":
-    Class = raw_input("> ")
-    if Class == "1" or Class == "Warrior":           #Warrior
-        print "Grand, fan of hitting and kicking huh? Your stats are below."
-        print ""
-        print "Strength = 8"
-        print "Dexterity = 4"
-        print "Wisdom = 2"
+def weapon():
+    weapon = Menu()
```

Comment:
While it's legal to do, I would be careful with using the same name to mean multiple things. It's easy to get confused about if you are talking about the `<type function>` weapon or the `<type Menu>`